### ISSUE TITLE
[bug] getDerivedHDAddress returns a serialized key

### ISSUE THREAD

**Issue by chromatic** [dogecoinfoundation/libdogecoin#217]
# Bug Report

**Current behavior:**

In at least version 0.1.3 and the branch for 0.1.4, `getDerivedHDAddress()` returns a key, not an address (returning the output of `getDerivedHDKeyByPath`.

**Expected behavior**

I'd like to receive an address instead.

**Comment by edtubbs** [comment]
Agreed, I think this is the original commit: https://github.com/dogecoinfoundation/libdogecoin/commit/9e4d7a645fc10d461aee5b651543b4704a73b386. 

With `derive_bip44_extended_key` you can extend to any level and then use `dogecoin_hdnode_get_p2pkh_address` to generate addresses.

**Comment by chromatic** [comment]
I can work around that in 0.1.3 now--can I rely on that version of the library providing those two functions you mentioned as stable and accessible?

**Comment by edtubbs** [comment]
Yes, both are in `libdogecoin.h` 0.1.3, in `example` and `bip44_tests`.

**Comment by chromatic** [comment]
I've pushed a new branch on my wrapper repository. If anyone reading this has the time and inclination to review the behavior, I welcome comments.

https://github.com/chromatic/Finance-libdogecoin/tree/libdogecoin-0.1.3-dev

**Comment by xanimo** [comment]
> I've pushed a new branch on my wrapper repository. If anyone reading this has the time and inclination to review the behavior, I welcome comments.
> 
> https://github.com/chromatic/Finance-libdogecoin/tree/libdogecoin-0.1.3-dev

i don't know perl but as far as i can tell it looks proper. was there an issue with the work around @edtubbs suggested?

**Comment by raffecat** [comment]
With regards to this, `getDerivedHDAddress` was the old way to walk down the tree of derived HD addresses, before `derive_bip44_extended_key` was added, and therefore it makes sense that it returns the full extended key (otherwise you wouldn't be able to derive any of its children.)

For backwards compatibility, I would suggest we don't change the existing functionality of this function (or any function), but rather expose new functions that are more convenient, such as `derive_bip44_extended_key` as @edtubbs suggested.

**Comment by edtubbs** [comment]
agree with @raffecat.. keep `getDerivedHDAddress` returning an extended key for backward compatibility. wrappers `getDerivedHDAddressByPath` and `getDerivedHDAddressAsP2PKH` already exist, plus mnemonic and encrypted variants.
